ArrtSource/Rendering/ArrSession.cpp

```diff
@@ -465,6 +474,70 @@ void ArrSession::ConfigureSessionPropertiesUpdateTimer()
     }
 }
 
+void ArrSession::CheckEntityBounds(RR::ApiHandle<RR::Entity> entity)
+{
+    entity->QueryWorldBoundsAsync([this](RR::Status status, RR::Bounds bounds)
+                                  {
+                                      if (status == RR::Status::OK && bounds.IsValid())
+                                      {
+                                          // pass the information to the main thread, because we want to interact with the GUI
+                                          QMetaObject::invokeMethod(QApplication::instance(), [this, bounds]()
+                                                                    { CheckEntityBoundsResult(bounds); });
+                                      }
+                                  });
+}
+
+void ArrSession::CheckEntityBoundsResult(RR::Bounds bounds)
+{
+    const float maxX = (float)bounds.Max.X;
+    const float minX = (float)bounds.Min.X;
+    const float maxY = (float)bounds.Max.Y;
+    const float minY = (float)bounds.Min.Y;
+    const float maxZ = (float)bounds.Max.Z;
+    const float minZ = (float)bounds.Min.Z;
+
+    const float width = maxX - minX;
+    const float height = maxY - minY;
+    const float depth = maxZ - minZ;
+
+    const float smallSize = 0.1f;
+    const float largeSize = 10.0f;
+    const float largeOffset = 10.0f;
+
+    const bool isLarge = (width > largeSize || height > largeSize || depth > largeSize);
+    const bool isSmall = (width < smallSize || height < smallSize || depth < smallSize);
+    const bool isFarAway = minX > largeOffset || minY > largeOffset || minZ > largeOffset ||
```

Comment:
nit: can be simplified: if (absf(minX) > largeOffset ...

Reply:
I don't think so, because it has to look at different variables.

Reply:
yes, but only 3 checks instead of 6

Reply:
oh wait